packages/core/src/utils/GroupGraph.ts

```diff
@@ -0,0 +1,62 @@
+import { Shape, ShapeAD } from "../types/shape";
+import Graph from "./Graph";
+
+export type GroupGraph = Graph<
+  string, // shape name
+  number // shape layer
+>;
+
+export const makeGroupGraph = <T extends Shape | ShapeAD>(
+  shapes: T[]
+): GroupGraph => {
```

Comment:
Why is this generic?

```suggestion
export const makeGroupGraph = (shapes: ShapeAD[]): GroupGraph => {
```

Reply:
Resolved - it used to be necessary, but not anymore.